### PR TITLE
Feat/init

### PR DESCRIPTION
**Pull request by mzahor**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[*.yml]
```
Comment (nirsky) [inline .editorconfig]: Not sure we need this file committed?

File: README.md [hunk 2]
```diff
@@ -1 +1,9 @@
 # genson-js
+
+**genson-js** is a user-friendly **JSON Schema** generator built in TypeScript/JavaScript.
```
Comment (nirsky) [inline README.md:3]: document available methods in readme

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './types';
+export * from './schema-builder';
```
Comment (nirsky) [inline src/index.ts]: do you wanna expose all funcs from this file?
like `unwrapSchema`?
I see it's exported..

File: tests/schema-builder.spec.ts [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+                    type: 'object',
+                    properties: { one: { type: 'integer' }, two: { type: 'string' } },
+                });
+            });
+        });
+
+        describe('nested array', () => {
+            it('should generate schema for nested arrays', async () => {
+                const schema = generateSchema([1, [2.1], [[3]]]);
+                expect(schema).toEqual({
```
Comment (nirsky) [inline tests/schema-builder.spec.ts:96]: maybe extract big expected payloads to different files? Will make it easier to follow to file

File: package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (nirsky) [inline package.json:1]: maybe add travis-ci to this repo?

File: README.md [hunk 6]
```diff
@@ -1 +1,9 @@
...
+
+.. note::
```
Comment (nirsky) [inline README.md]: you can you `>` (instead of ..) to get a cool styling fro notes

File: package.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    "name": "genson-js",
+    "version": "1.0.0",
+    "main": "dist/index.js",
+    "license": "MIT",
+    "scripts": {
+        "test": "jest",
+        "build": "tsc"
```
Comment (nirsky) [inline package.json]: Add prepublishOnly script that builds
also need `.npmignore` file

File: README.md (hunk 2 above)
Comment (mzahor) [reply README.md:3]: yep, that whole publish to npm part is yet to be done (including ci, readme, npm stuff)

File: src/index.ts (hunk 3 above)
Comment (mzahor) [reply src/index.ts]: Yeah, forgot about it, thanks.

File: tests/schema-builder.spec.ts (hunk 4 above)
Comment (mzahor) [reply tests/schema-builder.spec.ts:96]: extracted to snaphots and files.

File: package.json (hunk 5 above)
Comment (mzahor) [reply package.json:1]: will do, that's on the list

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,8 @@
+node_modules
+dist
+.DS_Store
+.idea
+.vscode
+yarn-error.log
+package-lock.json
```
Comment (habmic) [inline .gitignore]: package-log and yarn-error together are a bit strange?

File: package.json [hunk 9]
```diff
@@ -0,0 +1,50 @@
+{
+    "name": "genson-js",
```
Comment (habmic) [inline package.json:2]: I think we should add tags for search engine

**Review by habmic — COMMENTED** [review]
One thing I'm not sure about, generate schema sounds like we are creating a new schema. 
But we actually inferring it from an object, do you think we need to change the name / docs?

**Comment by mzahor** [comment]
@habmic `createSchema` maybe?

File: README.md (hunk 6 above)
Comment (mzahor) [reply README.md]: done

File: .gitignore (hunk 8 above)
Comment (mzahor) [reply .gitignore]: removed

**Comment by mzahor** [comment]
@habmic Fixed all comments, also some stuff that I found during testing. LMK if it's good for v0.0.1

**Comment by mzahor** [comment]
@habmic @nirsky Guys, I'd like to release 0.0.1, let me know if you have any other comments.

**Comment by habmic** [comment]
Yeah, I think we can go ahead and release it

**Pull request merged**